scripts/staging/fm-regression.dml

```diff
@@ -0,0 +1,169 @@
+      for (i in 1:iters) {
+        # Get the next batch
+        beg = ((i-1) * batch_size) %% N + 1
+        end = min(N, beg + batch_size - 1)
+        X_batch = X[beg:end,]
+        y_batch = y[beg:end,]
+
+        # 3.Send inputs through fm::forward
+        out = fm::forward(X_batch, w0, W, V);
+
+        # 4.compute gradients from a loss l2_loss::backward
+        dout = l2_loss::backward(out, y_batch)# (predictions, targets)
+
+        # Compute loss & accuracy for training & validation data every 100 iterations.
+        if (i %% 100 == 0) {
+          # Compute training loss & accuracy
+          #loss_data = cross_entropy_loss::forward(sfy, y_batch)
+          loss_data  = l2_loss::forward(out, y_batch)
+          loss_reg_w0 = l2_reg::forward(w0, lambda)
+          loss_reg_W  = l2_reg::forward(W , lambda)
+          loss_reg_V  = l2_reg::forward(V , lambda)
+          loss = loss_data + loss_reg_w0 + loss_reg_W + loss_reg_V
+          accuracy = ( mean( (dout - y_batch)^2 ) )^ 0.5
+
+
+          # Compute validation loss & accuracy
+          probs_val = predict(X_val, w0, W, V)
+          loss_val = l2_loss::forward(probs_val, y_val)
+          accuracy_val = ( mean( (probs_val - y_val)^2 ) )^0.5
+
+          # Output results
+          print("Epoch: " + e + ", Iter: " + i + ", Train Loss: " + loss + ", Train Accuracy: "
+                 + accuracy + ", Val Loss: " + loss_val + ", Val Accuracy: " + accuracy_val)
+        }
+
+        # 5.Send the above result through fm::backward
+        [dw0, dW, dV] = fm::backward(dout, X_batch, w0, W, V);
+
+        # 6.Call adam::update for all parameters
+        [w0,mw0,vw0] = adam::update(w0, dw0, lr, beta1, beta2, epsilon, t, mw0, vw0);
+        [W, mW, vW]  = adam::update(W, dW, lr, beta1, beta2, epsilon, t, mW, vW );
+        [V, mV, vV]  = adam::update(V, dV, lr, beta1, beta2, epsilon, t, mV, vV );
+
+      }
+    }
+}
+
+predict = function(matrix[double] X, matrix[double] w0, matrix[double] W, matrix[double] V)
+    return (matrix[double] out) {
+  /*
+   * Computes the predictions for the given inputs.
+   *
+   * Inputs:
+   *  - X : n examples with d features, of shape (n, d).
+   *  - w0, W, V : trained model parameters.
+   *
+   * Outputs:
+   *  - out : target vector, y.
+   */
+
+    # 1.Send inputs through fm::forward
+    out = fm::forward(X, w0, W, V);
+
+}
+
+eval = function(matrix[double] probs, matrix[double] y)
+    return (double loss, double accuracy) {
+   /*
+    * Computes loss and accuracy.
+    */
+
+    # just compute loss with log_loss::forward(probs, y)
+    loss = l2_loss::forward(probs, y);
+
+    # compute accuracy
+    sqr_mean = mean( (probs - y)^2 )
+    accuracy = (sqr_mean)^0.5
+    
+}
+
+# can the following be deleted, before committing?
+X = rand(rows=1000, cols=4); n = 1000; d = 4; k=2;
+y = rand(rows=n, cols=1);
+X_val = rand(rows=100, cols=4);
+y_val = rand(rows=100, cols=1);
+
+[w0, W, V] = train(X, y, X_val, y_val);
+print(as.scalar(V[3,1]));
```

Comment:
I would place all this into a `fm-regression-train-dummy.dml` script, similar to [`https://github.com/apache/systemml/blob/master/scripts/nn/examples/mnist_lenet_distrib_sgd-train-dummy-data.dml`](https://github.com/apache/systemml/blob/master/scripts/nn/examples/mnist_lenet_distrib_sgd-train-dummy-data.dml) (ignore the distributed stuff).